### PR TITLE
Feature/dto

### PR DESCRIPTION
**Pull request by picksitquick**
@aprajshekhar please review
Fixes #

## Proposed Changes

  - Added implementation of Mapper in Gamedoora UserService based on the changes made on Gamedoora Models.
  - Made changes in pom.xml based on run configs
  - APIs now return Dto instead of Dao
  - Please provide any feedback regarding any changes to be made.

### CODE REVIEW

File: src/main/java/com/gamedoora/backend/userservices/assembler/RolesServicesAssembler.java [hunk 1]
```diff
@@ -17,16 +17,11 @@ public class RolesServicesAssembler {
 
   @Autowired private RolesRepository rolesRepository;
 
-  public RoleDTO createRoles(RoleDTO rolesDto) {
+  @Autowired private RolesMapper rolesMapper;
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/assembler/RolesServicesAssembler.java:20]: Let's start using setter based autowiring.

File: src/main/java/com/gamedoora/backend/userservices/assembler/RolesServicesAssembler.java [hunk 2]
```diff
@@ -52,11 +47,11 @@ public void deleteAllRoles() {
     rolesRepository.deleteAll();
   }
 
-  public List<Roles> getAllRoles(String name) {
-    List<Roles> roles = new ArrayList<>();
-    if (name == null) rolesRepository.findAll().forEach(roles::add);
-    else rolesRepository.findByNameContaining(name).forEach(roles::add);
+  public List<RoleDTO> getAllRoles(String name) {
+    List<RoleDTO> roleDTOList = new ArrayList<>();
+    if (name == null) rolesRepository.findAll().forEach(roles -> roleDTOList.add(rolesMapper.roleToRoleDto(roles)));
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/assembler/RolesServicesAssembler.java:52]: Using getters would be a better approach.

File: src/main/java/com/gamedoora/backend/userservices/assembler/SkillsServicesAssembler.java [hunk 3]
```diff
@@ -21,19 +18,14 @@ public class SkillsServicesAssembler {
   @Autowired
   private SkillsRepository skillsRepository;
 
-  public SkillsDTO createSkills(SkillsDTO SkillsDto) {
+  @Autowired
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/assembler/SkillsServicesAssembler.java:21]: Let's start using setter based autowiring.

File: src/main/java/com/gamedoora/backend/userservices/assembler/SkillsServicesAssembler.java [hunk 4]
```diff
@@ -57,13 +49,13 @@ public void deleteAllSkills() {
...
   }
 
-  public List<Skills> getAllSkills(String name) {
-    List<Skills> skills = new ArrayList<Skills>();
+  public List<SkillsDTO> getAllSkills(String name) {
+    List<SkillsDTO> skillsDTOList = new ArrayList<>();
     if (name == null) {
-      skillsRepository.findAll().forEach(skills::add);
-      return skills;
+      skillsRepository.findAll().forEach(skills -> skillsDTOList.add(skillsMapper.skillsToSkillsDTO(skills)));
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/assembler/SkillsServicesAssembler.java:55]: Using getters would be a better approach.

**Review by aprajshekhar — CHANGES_REQUESTED** [review]
Thanks for the PR.  Just two suggestions:
- Using getters would be a better approach.
- Let's start using setter based autowiring.

**Comment by picksitquick** [comment]
@aprajshekhar 
I have made the requested changes, private fields now uses getter/setters.
Please review it, let me know if any changes are needed, or else we are good to merge this PR

**Review by aprajshekhar — APPROVED** [review]
Thanks for the changes. Approved and merging

**Pull request merged**